### PR TITLE
Add LSAT verification logic, Implement response logic for valid LSAT

### PR DESCRIPTION
**Pull request by DhananjayPurohit**
Added LSAT verification logic (signature verification and paymentHash should be equal to hash of preimage) and also implemented response logic for valid LSAT

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -22,11 +25,33 @@ type Service struct {
 func (svc *Service) getProtectedResource(c *gin.Context) {
 	authField, authFieldIsPresent := c.Request.Header["Authorization"]
```
Comment (bumi) [inline main.go:26]: maybe we can parse this header and extract the macaroon and the preimage in a utils function.

e.g. 
```
macaroon, preimage, err = utils.parseLsatHeader(c.Request.Header["Authorization"])
```
then we don't need to do the TrimSpace and such in the different methods.

File: main.go [hunk 2]
```diff
@@ -7,12 +7,15 @@ import (
 	"net/http"
 	"os"
 	"proxy/lnd"
-	"proxy/macaroon"
+	"proxy/lsat"
+	mintMacaroon "proxy/macaroon"
```
Comment (kiwiidb) [inline main.go:11]: better alias would be `macaroonutils`

File: main.go [hunk 3]
```diff
@@ -22,11 +25,33 @@ type Service struct {
 func (svc *Service) getProtectedResource(c *gin.Context) {
 	authField, authFieldIsPresent := c.Request.Header["Authorization"]
 
+	var IsValidMacaroon bool = false
```
Comment (kiwiidb) [inline main.go:28]:
```
isValidMacaroon := false
```

File: main.go [hunk 4]
```diff
@@ -35,15 +60,27 @@ func (svc *Service) getProtectedResource(c *gin.Context) {
...
 			return
 		}
 
 		c.Writer.Header().Set("WWW-Authenticate", fmt.Sprintf("LSAT macaroon=%v, invoice=%v", macaroonString, invoice))
 		c.String(http.StatusPaymentRequired, "402 Payment Required")
+		return
+	}
+
+	// Check valid LSAT
+	if authFieldIsPresent && IsValidLSAT {
```
Comment (kiwiidb) [inline main.go:81]: I'm pretty sure `authFieldIsPresent` always is `true` here. You could return the protected content on line 42, and then you can get rid of all the `if` blocks.

**Review by kiwiidb — CHANGES_REQUESTED** [review]
2 main things needed:

- inject the rootkey instead of hard-coding it
- refactor the conditional logic in the `getProtectedResource` endpoint

**Pull request merged**